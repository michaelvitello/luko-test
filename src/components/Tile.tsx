import React from 'react'
import { StyleSheet, View, Image, Text, Pressable } from 'react-native'
import { colors } from '../theme/colors'
import { fonts } from '../theme/fonts'

interface TileProps {
    image: string
    price: string | number
    title: string
    handleItemPress: () => void
}

const Tile = ({ image, price, title, handleItemPress }: TileProps): JSX.Element => {
    return (
        <Pressable
            onPress={handleItemPress}
            pressRetentionOffset={10}
            style={({ pressed }) => [{ opacity: pressed ? 0.5 : 1 }, styles.container]}
        >
            <Image source={{ uri: image }} style={styles.imageContainer} />
            <View style={styles.infoContainer}>
                <Text numberOfLines={2} style={styles.title}>
                    {title}
                </Text>
                <Text numberOfLines={1} style={styles.price}>{`${price} â‚¬`}</Text>
            </View>
        </Pressable>
    )
}

export default Tile

const styles = StyleSheet.create({
    container: {
        width: '92%', // Trick to get space in-between columns: columnWrapperStyle unsupported in Flashlist
        height: 250, // TODO: Better calculation
        borderRadius: 18,
        marginBottom: '8%',
        overflow: 'hidden',
        borderWidth: 0.3,
        borderColor: 'blue',
    },
    imageContainer: {
        resizeMode: 'cover',
        height: '50%',
        width: '100%',
    },
    infoContainer: {
        width: '100%',
        height: '50%',
        justifyContent: 'space-around',
        paddingLeft: 15,
        paddingRight: 15,
    },
    title: {
        fontFamily: fonts.bold,
        fontSize: 24,
    },
    price: {
        fontFamily: fonts.bold,
        fontSize: 15,
        color: colors.mainGrey,
    },
})
