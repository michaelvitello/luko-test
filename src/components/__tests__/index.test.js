import React from 'react'
import renderer from 'react-test-renderer'
import Tile from '../Tile'

test('renders Tile correctly', () => {
    const tree = renderer.create(<Tile />).toJSON()
    expect(tree).toMatchSnapshot()
})
