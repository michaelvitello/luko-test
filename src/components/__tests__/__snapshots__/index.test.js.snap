// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "borderColor": "blue",
        "borderRadius": 18,
        "borderWidth": 0.3,
        "height": 250,
        "marginBottom": "8%",
        "overflow": "hidden",
        "width": "92%",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": "50%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "height": "50%",
        "justifyContent": "space-around",
        "paddingLeft": 15,
        "paddingRight": 15,
        "width": "100%",
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontFamily": "Circular-Bold",
          "fontSize": 24,
        }
      }
    />
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#C0BEB8",
          "fontFamily": "Circular-Bold",
          "fontSize": 15,
        }
      }
    >
      undefined â‚¬
    </Text>
  </View>
</View>
`;
